lib/spack/spack/test/spec_syntax.py

```diff
@@ -818,26 +822,111 @@ def test_redundant_spec(query_str, text_fmt, database):
         ("x platform=test platform=test", spack.spec.DuplicateArchitectureError),
         ("x os=fe platform=test target=fe os=fe", spack.spec.DuplicateArchitectureError),
         ("x target=be platform=test os=be os=fe", spack.spec.DuplicateArchitectureError),
+    ],
+)
+def test_error_conditions(text, exc_cls):
+    with pytest.raises(exc_cls):
+        SpecParser(text).next_spec()
```

Comment:
We don't need to split the test, since we can mark individual input, see [here](https://docs.pytest.org/en/7.1.x/example/markers.html#marking-individual-tests-when-using-parametrize)

Reply:
Oh you can make `xfail` conditional, that cleans things up a lot!